src/wrapper/data_chunk.ml

```diff
@@ -29,6 +29,47 @@ let get_opt t type_ idx =
 
 let free t ~here = Resource.free t.data_chunk ~here
 
+let column_count t =
```

Comment:
Instead of relying on an exception to figure out the number of columns, the function should take an argument indicating it.